web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala

```diff
@@ -95,23 +96,27 @@ import scala.util.{Failure, Success}
         )
     )
 
-    val error = formData.firstValidationError.map { text =>
-      val emailNotVerified = "The email is not verified, check your spam folder if you don't see the email."
+    def resendVerifyEmailButton(text: String) = {
+      val emailNotVerified = ErrorMessages.emailNotVerified
+
+      Fragment(text match {
```

Comment:
I'm not sure if this Fragment is necessary

Reply:
It's necessary because if I remove the Fragment the app throws a `type mismatch` error

Reply:
I can remove `case _ => None` and it won't throw the error without the Fragment

Reply:
nvm, the page throws me this error if I remove the Fragment and `case _ => None`
<img src="https://user-images.githubusercontent.com/80025691/184004327-dc23f1b0-94e5-48a1-9546-11bff842b299.png" width="500">

Reply:
The problem with removing `case _` is that it can cause a `MatchError` at runtime if `text != emailNotVerified` because we aren't handling the case when the two variables are different

I can think of two ways to solve this:
1. Return a Option[ReactElement] type
```
text match {
  case `emailNotVerified` =>
          Some(mui
            .Button(texts.resendEmail)
            .variant(muiStrings.text)
            .color(muiStrings.primary)
            .onClick(_ => history.push("/resend-verify-email")))
  case _ => None
}
```

2. Return a ReactElement type (the Fragment is used so group components together, so it isn't visible to the user)
```
text match {
  case `emailNotVerified` =>
          mui
            .Button(texts.resendEmail)
            .variant(muiStrings.text)
            .color(muiStrings.primary)
            .onClick(_ => history.push("/resend-verify-email")))
  case _ => Fragment()
}
```

I'd personally use the second approach but if you find a better way to do this without causing a `MatchError` at runtime feel free to use. On my experience that's what I use most of the time